pytorch_lightning/trainer/trainer.py

```diff
@@ -527,7 +522,15 @@ def train(self):
                     return
 
                 # update LR schedulers
-                self.optimizer_connector.update_learning_rates(interval='epoch')
+                self.optimizer_connector.update_learning_rates(
+                    interval='epoch',
+                    opt_indices=[
+                        opt_idx
+                        for opt_idx, _ in self.train_loop.get_optimizers_iterable(
+                            batch_idx=(self.total_batch_idx - 1)
+                        )  # Select the optimizers which were used in the last batch of the epoch
```

Comment:
Need to get the optimizers that were used in the last batch of the epoch here. Without the `- 1`, `get_optimizers_iterable` will return optimizers that will be used in the next batch since the epoch has already finished.

Reply:
Also an interesting situation would be when multiple schedulers are used during the epoch. For eg, if scheduler A is used during the first half of the epoch and scheduler B during the second half, the current code will only update scheduler B (assuming both use `epoch` interval). Do you folks think this scenario is likely?

In order to solve this, we'll need to get all the optimizers used in the epoch and pass their indices to `opt_indices`. I can update the code and add a test for that scenario as well if it needs to be addressed.

Reply:
I don't think this is a valid case. Also never seen this before.

Reply:
Ok then do you suggest to leave this as is?

Reply:
+1, LightningOptimizer should handle the schedulers too. Just wondering how epoch-level schedulers will work with that.